scripts/zones/RoMaeve/Zone.lua

```diff
@@ -3,60 +3,55 @@
-require("scripts/globals/conquest");
-require("scripts/globals/missions");
-require("scripts/globals/npc_util");
-require("scripts/globals/weather");
-require("scripts/globals/status");
+local ID = require("scripts/zones/RoMaeve/IDs")
+require("scripts/globals/conquest")
+require("scripts/globals/missions")
+require("scripts/globals/npc_util")
+require("scripts/globals/weather")
+require("scripts/globals/status")
 -----------------------------------
 
 function onInitialize(zone)
-    local newPosition = npcUtil.pickNewPosition(ID.npc.BASTOK_7_1_QM, ID.npc.BASTOK_7_1_QM_POS, true);
-    GetNPCByID(ID.npc.BASTOK_7_1_QM):setPos(newPosition.x, newPosition.y, newPosition.z);
-end;
+    local newPosition = npcUtil.pickNewPosition(ID.npc.BASTOK_7_1_QM, ID.npc.BASTOK_7_1_QM_POS, true)
+    GetNPCByID(ID.npc.BASTOK_7_1_QM):setPos(newPosition.x, newPosition.y, newPosition.z)
+end
 
 function onConquestUpdate(zone, updatetype)
     tpz.conq.onConquestUpdate(zone, updatetype)
-end;
+end
 
 function onZoneIn(player,prevZone)
-    local cs = -1;
-    if (player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0) then
-        player:setPos(-0.008,-33.595,123.478,62);
+    local cs = -1
+    if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
+        player:setPos(-0.008,-33.595,123.478,62)
     end
-    if (player:getCurrentMission(WINDURST) == tpz.mission.id.windurst.VAIN and player:getCharVar("MissionStatus") ==1) then
-        cs = 3; -- doll telling "you're in the right area"
+    if player:getCurrentMission(WINDURST) == tpz.mission.id.windurst.VAIN and player:getCharVar("MissionStatus") ==1 then
+        cs = 3 -- doll telling "you're in the right area"
     end
-    return cs;
-end;
+    return cs
+end
 
 function onRegionEnter(player,region)
-end;
+end
 
-function onGameDay()
-    if (IsMoonFull() and GetNPCByID(ID.npc.MOONGATE_OFFSET):getWeather() == tpz.weather.SUNSHINE) then
-        for i = ID.npc.MOONGATE_OFFSET, ID.npc.MOONGATE_OFFSET+7 do
-            GetNPCByID(i):openDoor(432);
-        end
-    end
-end;
+function onGameHour(zone)
+    local hour = VanadielHour()
 
-function onZoneWeatherChange(weather)
-    if (weather ~= tpz.weather.SUNSHINE and GetNPCByID(ID.npc.MOONGATE_OFFSET):getAnimation() ~= tpz.anim.CLOSE_DOOR) then
+    if IsMoonFull() and hour == 18 then
         for i = ID.npc.MOONGATE_OFFSET, ID.npc.MOONGATE_OFFSET+7 do
-            GetNPCByID(i):setAnimation(tpz.anim.CLOSE_DOOR);
+            GetNPCByID(i):openDoor(1728)
+            GetNPCByID(i):untargetable(true)
         end
-    elseif (weather == tpz.weather.SUNSHINE and IsMoonFull() == true and VanadielHour() < 3) then -- reactivate things for the remainder of the time until 3AM
-        local moonMinRemaining = math.floor(432 * (180 - VanadielHour() * 60 + VanadielMinute())/180) -- 180 minutes (ie 3AM) subtract the time that has passed since midnight
+    elseif IsMoonFull() and hour == 6 then
```

Comment:
Will the moon still be "full" after the final night? We wouldn't want to accidentally leave the doors untargetable. 😉

Reply:
change to
    elseif IsMoonFull() then
as
    elseif hour == 6 and GetNPCByID(ID.npc.MOONGATE_OFFSET):untargetable() then
didnt work for some reason even tho it looks legit (i just didnt want to be valid on every single vanaday).

Reply:
We might be able to set up a timer when we _open_ the door, but it's probably not worth it - setting it to false once a real-life hour isn't expensive (and might actually be cheaper than adding an additional timer).